notes/stv/20221014-test-benchmarker-01.txt

```diff
@@ -0,0 +1,325 @@
+	> sys	2m45.715s
+
+
+# -----------------------------------------------------
+# Import our live users.
+#[root@ansibler]
+
+    source /deployments/zeppelin/bin/create-user-tools.sh
+
+    import-test-users
+
+# -----------------------------------------------------
+# Start NGINX so that we can access using public IP.
+#[fedorat@zeppelin]
+
+    sudo service nginx start
+
+
+    cat > /tmp/testprog.py << EOF
+import sys
+from aglais_benchmark import AglaisBenchmarker
+AglaisBenchmarker(
+    "/deployments/zeppelin/test/config/quick.json",
+    "/tmp/test-users.json",
+    "/tmp/",
+    "http://128.232.227.167"
+    ).run(
+        concurrent=False,
+        users=1
+        )
+EOF
+
+    python3 /tmp/testprog.py
+
+[
+	[{
+		"name": "GaiaDMPSetup",
+		"result": "PASS",
+		"outputs": {
+			"valid": true
+		},
+		"messages": [],
+		"time": {
+			"result": "SLOW",
+			"elapsed": "45.45",
+			"expected": "45.00",
+			"percent": "1.01",
+			"start": "2022-10-14T15:42:28.059656",
+			"finish": "2022-10-14T15:43:13.512597"
+		},
+		"logs": ""
+	}, {
+		"name": "Mean_proper_motions_over_the_sky",
+		"result": "PASS",
+		"outputs": {
+			"valid": true
+		},
+		"messages": [],
+		"time": {
+			"result": "SLOW",
+			"elapsed": "123.49",
+			"expected": "55.00",
+			"percent": "124.53",
+			"start": "2022-10-14T15:43:13.512698",
+			"finish": "2022-10-14T15:45:17.005341"
+		},
+		"logs": ""
+	}, {
+		"name": "Source_counts_over_the_sky.json",
+		"result": "PASS",
+		"outputs": {
+			"valid": true
+		},
+		"messages": [],
+		"time": {
+			"result": "SLOW",
+			"elapsed": "51.48",
+			"expected": "22.00",
+			"percent": "134.00",
+			"start": "2022-10-14T15:45:17.005456",
+			"finish": "2022-10-14T15:46:08.486087"
+		},
+		"logs": ""
+	}, {
+		"name": "Library_Validation.json",
+		"result": "ERROR",
+		"outputs": {
+			"valid": true
+		},
+		"messages": [],
+		"time": {
+			"result": "FAST",
+			"elapsed": "7.67",
+			"expected": "60.00",
+			"percent": "-87.21",
+			"start": "2022-10-14T15:46:08.486807",
+			"finish": "2022-10-14T15:46:16.158120"
+		},
+		"logs": "Fail to execute line 6: import hdbscan\nTraceback (most recent call last):\n  File \"/tmp/1665762183782-0/zeppelin_python.py\", line 158, in <module>\n    exec(code, _zcUserQueryNameSpace)\n  File \"<stdin>\", line 6, in <module>\n  File \"/usr/local/lib64/python3.7/site-packages/hdbscan/__init__.py\", line 1, in <module>\n    from .hdbscan_ import HDBSCAN, hdbscan\n  File \"/usr/local/lib64/python3.7/site-packages/hdbscan/hdbscan_.py\", line 334, in <module>\n    algorithm='best', memory=Memory(cachedir=None, verbose=0),\nTypeError: __init__() got an unexpected keyword argument 'cachedir'"
+	}]
+]
+
+
+    cat > /tmp/testprogmulti.py << EOF
+import sys
+from aglais_benchmark import AglaisBenchmarker
+AglaisBenchmarker(
+    "/deployments/zeppelin/test/config/quick.json",
+    "/tmp/test-users.json",
+    "/tmp/",
+    "http://128.232.227.167"
+    ).run(
+        concurrent=True,
+        users=2
```

Comment:
Technically 2 > 1, but still. This is a very very minimal set of tests.
The reason for running tests is to find bugs. Not just to tick the box.

Reply:
I've run some additional tests which should be included in the next commit